### DISCUSSION TITLE
Correct way to cancel job and trigger, scheduler.UnscheduleJob(_tKey); doesn't fully work.

### DISCUSSION THREAD

**Discussion by CrazyMoka** [quartznet/quartznet#1711] [Q&A] [answered]
Hello
I'm trying to implement a way to cancel a currently running job if its running or waiting to run.
Currently I'm only running scheduler.UnscheduleJob(_myTriggerKey); 
However, it isn't stopping the job. The trigger was removed from the DB but the job is still running.

I also sometimes run into the situation where running scheduler.UnscheduleJob(_tKey); doesn't remove the trigger from the DB and it doesn't get removed sometimes on its own.  Trying to figure out how to haddle this.

Thank you

**Comment by lahma** [comment, marked as answer]
You can use `scheduler.Interrupt()`, there's also [a sample in examples project](https://github.com/quartznet/quartznet/tree/main/src/Quartz.Examples/07_InterrupInProgressJobs)

https://www.quartz-scheduler.net/documentation/faq.html#how-do-i-stop-a-job-that-is-currently-executing
